linux/src/async_socket_linux.c

```diff
@@ -115,6 +116,9 @@ static void event_complete_callback(void* context, COMPLETION_PORT_EPOLL_ACTION
     else
     {
         ASYNC_SOCKET_IO_CONTEXT* io_context = (ASYNC_SOCKET_IO_CONTEXT*)context;
```

Comment:
>context [](http://example.com/codeflow?start=72&length=7)

What happens if you insert here a Sleep() and while the sleep is executing, you call "_close"?

Wouldn't that lead to the same helgrind error?

I will be waiting.

Reply:
I could test to see if the module is in the open state, and then I'll be sure you can execute it.  Let me test that first.